src/platform/src/test/java/org/geoserver/platform/GeoServerResourceLoaderTest.java

```diff
@@ -124,5 +132,28 @@ public void testLookupRequireMissingFileServlet() {
                 + "From: Servlet context parameter GEOSERVER_REQUIRE_FILE: does-not-exist");
         GeoServerResourceLoader.lookupGeoServerDataDirectory(context);
     }
+    
+    @Test
+    public void testSetBaseDirectory() throws IOException {
+        GeoServerResourceLoader loader = new GeoServerResourceLoader();
+        assertNull(loader.getBaseDirectory());
+        assertEquals(ResourceStore.EMPTY, loader.getResourceStore());
+        
+        TemporaryFolder tempFolder = new TemporaryFolder();
```

Comment:
TemporaryFolder needs to be set as a field with a @Rule to be cleaned up at test end, or not?
http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html